server.js

```diff
@@ -1,33 +1,73 @@
-import express from 'express'
+import express from 'express' //node framework -to create the backend-server
 import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import booksData from './data/books.json'
 
 // Defines the port the app will run on. Defaults to 8080, but can be 
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080 //fundamental settings
+const app = express() //intiating the express app
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(express.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+const users = [
+  { id: 1, name: 'Rosanna', age: 35 },
+  { id: 2, name: 'Anton', age: 36 },
+  { id: 3, name: 'Freyja', age: 4 },
+  { id: 4, name: 'Albin', age: 2 },
+]
+
+// This is our first endpoint //if remove this we get "Cannot GET response in browser" it's built in by express. The server is running but there aren't any information.
+app.get('/', (req, res) => { //This app.get method takes two arguments: path and maybe users or movies. //The second argument is a call back function that communicates with the front end. 
+  //The request (req) is what the frontend sends to the backend. The response (res) is what the backend send to the frontend.
+  res.send('Hello from Rosanna to you!')
 })
 
-// Start the server
+// get a list of users
+app.get('/users', (req, res) => {
+  res.json(users)
+})
+
+// get a list of the books (from json file)
+app.get('/books', (req, res) => {
+  res.json(booksData)
+})
+
+// get a specific book based on its id, using param
+app.get('/books/:bookid', (req, res) => { //this endpoint is about finding the book with the specific Id
+  const { bookid } = req.params
+
+  const book = booksData.find(book => book.bookID == bookid)
+
+  if (!book) {
+    res.status(404).send('No book found with that id')
+  } else {
+    res.json(book)
+  }
+})
+
+
+// get a specific rating on book higher then route parameter
+app.get('/books3/:rating', (req, res) => {
+  const { rating } = req.params
+
+  const books = booksData.filter(book => book.average_rating >= rating)
```

Comment:
Nice! Like that the user of the API can seach for books with a certain rating, really useful!!!!